pkg/controller/core/workload_controller.go

```diff
@@ -192,6 +193,33 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 			return ctrl.Result{}, err
 		}
 
+		if features.Enabled(features.DynamicallySizedJobs) && ptr.Deref(wl.Status.Admission.PodSetAssignments[1].Count, 0) != wl.Spec.PodSets[1].Count {
```

Comment:
Please put all of this inside a method.

Maybe `downSizeJobIfNecessary`

Reply:
why only check one podset? AFAIK, RayCluster can have multiple podsets.

Reply:
Checking for multiple podsets now